tls/s2n_resume.c

```diff
@@ -776,9 +809,17 @@ S2N_RESULT s2n_resume_encrypt_session_ticket(struct s2n_connection *conn, struct
     /* No keys loaded by the user or the keys are either in decrypt-only or expired state */
     RESULT_ENSURE(key != NULL, S2N_ERR_NO_TICKET_ENCRYPT_DECRYPT_KEY);
 
+    /* Generate unique per-ticket encryption key */
+    struct s2n_unique_ticket_key ticket_key = { 0 };
+    RESULT_CHECKED_MEMCPY(ticket_key.initial_key, key->aes_key, sizeof(key->aes_key));
+    struct s2n_blob info_blob = { 0 };
+    RESULT_GUARD_POSIX(s2n_blob_init(&info_blob, ticket_key.info, sizeof(key->aes_key)));
+    RESULT_GUARD(s2n_get_public_random_data(&info_blob));
+    RESULT_GUARD(s2n_resume_generate_unique_ticket_key(&ticket_key));
+
     /* Initialize AES key */
     struct s2n_blob aes_key_blob = { 0 };
-    RESULT_GUARD_POSIX(s2n_blob_init(&aes_key_blob, key->aes_key, sizeof(key->aes_key)));
+    RESULT_GUARD_POSIX(s2n_blob_init(&aes_key_blob, ticket_key.output_key, sizeof(key->aes_key)));
```

Comment:
The blob length should probably be set based on the new value, even though the sizes should be the same:
```suggestion
    RESULT_GUARD_POSIX(s2n_blob_init(&aes_key_blob, ticket_key.output_key, sizeof(ticket_key.output_key)));
```